src/JWT/Builder/JwtBuilder.cs

```diff
@@ -276,7 +282,7 @@ public string Encode(Type payloadType, object payload)
             EnsureCanEncode();
 
             var dic = payloadType.GetProperties(BindingFlags.Instance | BindingFlags.Public)

```

Comment:
Overall, I think this is not the rich approach, the library shouldn't try to reinvent the wheel, must somehow let the serializer itself to deal with the type of the payload.

Reply:
@hartmark what you think about this variant of the method?

```c#
  public string Encode(Type payloadType, object payload)
  {
      if (payloadType is null)
          throw new ArgumentNullException(nameof(payloadType));
      if (payload is null)
          throw new ArgumentNullException(nameof(payload));
      
      EnsureCanEncode();

      return _encoder.Encode(_jwt.Header, payload, _secrets?[0]);
  }
```

which passes the newly added tests too!

Reply:
The tests that just do .Encode without .AddClaim works

For example Encode_Should_Return_Token_With_Custom_Extra_Headers_Full_Payload_And_Claims fails as it adds the claim "ExtraClaim" first and that gets overwritten by .Encode method.